.github/workflows/ci.yml

```diff
@@ -32,6 +32,9 @@ jobs:
       - name: Lint
         run: npm run lint
 
+      - name: Benchmark
```

Comment:
Do we really want to run benchmark tests at each push ?
This is a thing that I would handle separately with a benchmark triggered manually or only on the master branch push.
Moreover we could need a strategy to push benchmarking results when we npm run lint.

Reply:
I do not know if GitHub CI permit to define manually triggered steps. 

I think the benchmark run on PR is valuable, it can be changed to a performance regression test if a sensible threshold is defined that pool implementations should never be up to.